tests/test_extractors.py

```diff
@@ -91,3 +91,15 @@ def test_extract_from_directory(self, extractor, tmp_path, file):
     def test_extract_from_nonsense(self, extractor):
         with pytest.raises(NotImplementedError):
             extractor.extract(8)
+
+    @pytest.mark.parametrize(
+        "source",
+        [
+            """
+                def foo():
+            """,
+        ],
+    )
+    def test_extract_with_syntax_error(self, extractor, file_path, file, caplog):
```

Comment:
**suggestion (testing):** Add assertions for the content of the warning message.

While the test currently checks that a warning is issued, it would be more comprehensive to also verify that the warning message contains the expected information about the syntax error.